utils/docker/dev_peer_cc-builder/Dockerfile

```diff
@@ -151,6 +158,8 @@ FROM common as cc-builder
 # import global vars
 ARG SGX_MODE
 ENV SGX_MODE=${SGX_MODE}
+# Note: the enclave.so does not depend on FPC_ATTESTATION_TYPE, so we only have to specialize on SGX_MODE
```

Comment:
Would you double check this please?

I think it's actually the opposite.
cc-builder builds the sdk, including sgxcclib in `ecc_enclave`, and it will build it with epid-unlinkable by default.
Then, cc-builder and its artifacts are used by the demo to build the enclave.so.
However, if I recall correctly, the demo skips the ercc stuff, so it does not fail.

Reply:
cc-builder definitely builds a attestation-type dependent sdk. But the attestation part should be all only in the untrusted part, not in the trusted code (i.e., it is backed statically into the invocation of `sgx_get_quote` in common/sgxcclib/common-sgxcclib.c`).  So it shouldn't `enclave.so`, the only build artifact we use `cc-builder` for.  I've also tested with both linkable and unlinkable with `make clobber all` (which runs real attestation in the non-interactive auction demo including cc-builder) and even tested with wrong credentials which also failed as expected.   So not sure what i've overlooked?   But maybe i could be a bit more explicit in the Note: for the reasoning?

Reply:
That's fine, the "cc-builder" created some confusion.
Summarizing: just as we do for any FPC chaincode (see the examples), we generate the `enclave.so` and we use the pre-built `ecc` and `libsgxcc`. In this particular case, cc-builder creates the `enclave.so`.